services/net/extract-quotes/ExtractQuotesManager.cs

```diff
@@ -0,0 +1,287 @@
+            !_notRunning.Contains(_consumer.Status) &&
+            _cancelToken != null && !_cancelToken.IsCancellationRequested)
+        {
+            _cancelToken.Cancel();
+        }
+    }
+
+    /// <summary>
+    /// Retrieve a file from storage and send to Microsoft Cognitive Services. Obtain
+    /// the report and update the content record accordingly.
+    /// </summary>
+    /// <param name="result"></param>
+    /// <returns></returns>
+    private async Task HandleMessageAsync(ConsumeResult<string, IndexRequestModel> result)
+    {
+        try
+        {
+            // The service has stopped, so to should consuming messages.
+            if (this.State.Status != ServiceStatus.Running)
+            {
+                this.Listener.Stop();
+                this.State.Stop();
+            }
+            else
+            {
+                await ProcessIndexRequestAsync(result);
+
+                // Inform Kafka this message is completed.
+                this.Listener.Commit(result);
+                this.Listener.Resume();
+
+                // Successful run clears any errors.
+                this.State.ResetFailures();
+                _retries = 0;
+            }
+        }
+        catch (Exception ex)
+        {
+            if (ex is HttpClientRequestException httpEx)
+            {
+                this.Logger.LogError(ex, "HTTP exception while consuming. {response}", httpEx.Data["body"] ?? "");
+                await this.SendEmailAsync("HTTP exception while consuming. {response}", ex);
+            }
+            else
+            {
+                this.Logger.LogError(ex, "Failed to handle message");
+                await this.SendEmailAsync("Failed to handle message", ex);
+            }
+            ListenerErrorHandler(this, new ErrorEventArgs(ex));
+        }
+        finally
+        {
+            if (State.Status == ServiceStatus.Running) Listener.Resume();
+        }
+    }
+
+    /// <summary>
+    /// Process the report request.
+    /// </summary>
+    /// <param name="request"></param>
+    /// <returns></returns>
+    private async Task ProcessIndexRequestAsync(ConsumeResult<string, IndexRequestModel> result)
+    {
+        this.Logger.LogDebug($"ProcessIndexRequestAsync:BEGIN:{result.Message.Key}");
+
+        this.Logger.LogInformation("Indexing content from Topic: {Topic}, Content ID: {Key}", result.Topic, result.Message.Key);
+        var model = result.Message.Value;
+
+        if (model.Action == IndexAction.Index || model.Action == IndexAction.Publish)
+        {
+            // TODO: Failures after receiving the message from Kafka will result in missing content.  Need to handle this scenario.
+            var content = await this.Api.FindContentByIdAsync(result.Message.Value.ContentId);
+            if (content != null)
+            {
+                // Update the unpublished content with the latest data and status.
+                await ExtractQuotesAsync(content);
```

Comment:
What are you going to do after extracting the quotes?  Update the database?  Will the frontend make a separate requests for quotes?

Reply:
Conceptually, when we extract quotes for a piece of content it would be associated with it's parent content in the database.  In order for the quotes to be indexed by elastic, we would have to trigger a re-index, which could re-trigger the quote extraction service. Might have to think about that...

If the quote extraction service is the sole source of truth for extracting/associating quotes with content, then there doesnt need to be any UI around adding quotes to content.  We just need to display the ones which were extracted.